src/qtpromise/qpromisehelpers_p.h

```diff
@@ -88,6 +88,61 @@ void connectDestroyedToReject(const QtPromise::QPromiseConnections& connections,
     });
 }
 
+template<typename T, typename U>
+struct AreArithmeticTypes
```

Comment:
True, we can possibly decide for `static_cast`. My original idea was to call the constructor To(From) for non-arithmetic types to allow conversions like QByteArray -> QString. Initially, I thought there should be a distinction: if we decided for uniform initialization like in `To{value}`, we would run into a problem with `double` to `int` conversion because it's a narrowing conversion, and the uniform initialization generates a warning [1]. As for static cast, I thought you had to define a conversion operator in user-defined classes, but apparently a non-explicit constructor with a single argument of the required type is called a _converting constructor_ and is good enough ([2, 3, 4]).

So we can stick to `static_cast`.

[1] https://godbolt.org/z/xKMdxW
[2] https://en.cppreference.com/w/cpp/language/cast_operator
[3] https://en.cppreference.com/w/cpp/language/converting_constructor
[4] https://godbolt.org/z/8GdKEG

Reply:
Thank you for the detailed answer, let's use `static_cast` and get rid of the special case.